sys/include/suit/v4/suit.h

```diff
@@ -91,74 +101,191 @@ typedef enum {
 } suit_v4_component_t;
 
 /**
  * @brief SUIT manifest struct
  */
 typedef struct {
-    cose_sign_t cose;   /**< COSE signature validation struct */
-    const uint8_t *buf; /**< ptr to the buffer of the manifest */
-    size_t len;         /**< length of the manifest */
-    uint32_t validated; /**< bitfield of validated policies */
-    uint32_t state;     /**< bitfield holding state information */
+    cose_sign_t cose;               /**< COSE signature validation struct */
+    const uint8_t *buf;             /**< ptr to the buffer of the manifest */
+    size_t len;                     /**< length of the manifest */
+    uint32_t validated;             /**< bitfield of validated policies */
+    uint32_t state;                 /**< bitfield holding state information */
 
+    /** List of components in the manifest */
     suit_v4_component_t components[SUIT_V4_COMPONENT_MAX];
     //CborValue components[SUIT_V4_COMPONENT_MAX];
-    unsigned components_len;
-    int component_current;
-    riotboot_flashwrite_t *writer;
+    unsigned components_len;        /**< Current number of components */
+    int component_current;          /**< Current component index */
+    riotboot_flashwrite_t *writer;  /**< Pointer to the riotboot flash writer */
+    /** Manifest validation buffer */
     uint8_t validation_buf[SUIT_COSE_BUF_SIZE];
-    cose_key_t *key;    /**< Ptr to the public key for validation */
-    char *urlbuf;
-    size_t urlbuf_len;
+    cose_key_t *key;                /**< Ptr to the public key for validation */
+    char *urlbuf;                   /**< Buffer containing the manifest url */
+    size_t urlbuf_len;              /**< Length of the manifest url */
 } suit_v4_manifest_t;
 
-#define SUIT_MANIFEST_HAVE_COMPONENTS 0x1
-#define SUIT_MANIFEST_HAVE_IMAGE 0x2
+/**
+ * @brief Mask used to determine if SUIT manifest contains components
+ */
+#define SUIT_MANIFEST_HAVE_COMPONENTS   (0x1)
+/**
+ * @brief Mask used to determine if SUIT manifest contains an image
+ */
+#define SUIT_MANIFEST_HAVE_IMAGE        (0x2)
 
 /**
  * @brief Parse a manifest
  *
  * @note The buffer is still required after parsing, please don't reuse the
  * buffer while the @p manifest is used
  *
- * @param   manifest    manifest context to store information in
- * @param   buf         buffer to parse the manifest from
- * @param   len         length of the manifest data in the buffer
+ * @param[in]   manifest    manifest context to store information in
+ * @param[in]   buf         buffer to parse the manifest from
+ * @param[in]   len         length of the manifest data in the buffer
  *
  * @return              SUIT_OK on parseable manifest
  * @return              negative @ref suit_v4_error_t code on error
  */
 int suit_v4_parse(suit_v4_manifest_t *manifest, const uint8_t *buf, size_t len);
 
+/**
+ * @brief Check a manifest policy
+ *
+ *
+ * @param[in]   manifest    manifest context to store information in
```

Comment:
Should this be `manifest context to check the policy for`?

Reply:
ah yes, copy-pasted from above...